sdk/typescript/src/module/introspector/test/case_convertor.spec.ts

```diff
@@ -33,5 +33,30 @@ describe("case convertor", function () {
       const result = convertToPascalCase("HelloWorld")
       assert.equal(result, "HelloWorld")
     })
+
+    it("should correctly handle numbers as spacers", function () {
+      const result = convertToPascalCase("hello123world")
+      assert.equal(result, "Hello123World")
+    })
+
+    it("should correctly handle number as first word", function () {
+      const result = convertToPascalCase("123helloworld")
+      assert.equal(result, "123Helloworld")
+    })
+
+    it("should correctly handle number as last word", function () {
+      const result = convertToPascalCase("helloWorld123")
+      assert.equal(result, "HelloWorld123")
+    })
+
+    it("should correctly handle special characters as spacers", function () {
+      const result = convertToPascalCase("hello world")
+      assert.equal(result, "HelloWorld")
+    })
+
+    it("should correctly handle mix cases", function () {
```

Comment:
Slightly curious how it handles something like `introspectionJSON` (where the second word is an acronym). Might be worth checking that that case also works (since those style of acronyms have caused other problems in other SDKs before).

Reply:
I'll add a case, nice catch

Reply:
Ah, this feels very similar to the corresponding go sdk/engine issue as https://github.com/dagger/dagger/issues/7941

We need to redo a lot of string-casing logic and make it entirely consistent everywhere. However, this is such a pain, and so completely breaking, I'm still not sure how we do it properly (which is why the work on https://github.com/dagger/dagger/pull/8122 stalled).

Reply:
Tests cases added, however, like we discussed on Discord, it will still fails on user's end because the engine generates `introspectionJson` instead of `introspectionJSON`

Reply:
Yup, SGTM, good to merge whenever. Let's just fix the immediate pain, we need to circle back round to this eventually one day anyways.